### PR TITLE
Add prim.svd, refs.linalg.svd, and refs.linalg.svdvals

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #81113
* __->__ #78616
* #78615
* #78350
* #80219

This is the first prim / ref added that has multiple returns.
There is an issue with `out_wrapper_multi` as currently implemented
(left a note). It assumes that the API is `svd(X, U=U, S=S, Vh=Vh)`,
when it's actually `svd(X, out=(U, S, Vh))`.

Even more, if we want to model PyTorch exactly, it should return a
`torch.return_types.svd`, rather than a `Tuple`.

There is an issue with
As per title

### CODE REVIEW

File: torch/_prims/__init__.py [hunk 1]
```diff
@@ -2467,6 +2473,73 @@ def _full_like_aten(
...
+        dtype=utils.corresponding_real_dtype(A.dtype),
+        device=A.device,
+    )
+
+    shape_Vh = batch + (n if full_matrices else k, n)
+    # The CPU backend returns V, but the cuSolver backend returns V^H
+    # TODO The MAGMA backend returns V, so this is wrong if used with the MAGMA backend
+    is_cuda = (A.device.type == "cuda")
+    strides_Vh = contig_strides(shape_Vh, row_major=is_cuda)
+    Vh = TensorMeta(shape=shape_Vh, strides=strides_Vh, dtype=A.dtype, device=A.device)
```
Comment (Lezcano) [inline torch/_prims/__init__.py:2518]: Not clear what to do here.

File: torch/_prims/__init__.py [hunk 2]
```diff
@@ -2467,6 +2473,73 @@ def _full_like_aten(
...
+    return U, S, Vh
+
+
+def _svd_aten(
+    A: TensorLikeType, *, full_matrices: bool
+) -> Tuple[Tensor, Tensor, Tensor]:
```
Comment (Lezcano) [inline torch/_prims/__init__.py:2524]: Not sure whether we should model here the `compute_uv` parameter. On the one hand, it is useful when computing `svdvals`, but only when no gradients are required. As such, this optimisation is not fully contemplated at the moment in PrimTorch, given that we are not considering gradients yet.
Should we add the `compute_uv` parameter here (as we do in `_linalg_svd` in PyTorch really)?

Comment (mruberry) [reply torch/_prims/__init__.py:2473]: Good thing we can always change our mind later -- let's try it without `compute_uv` for now

File: torch/_prims/__init__.py [hunk 3]
```diff
@@ -2416,6 +2422,73 @@ def _full_like_aten(
...
+    return torch.linalg.svd(A, full_matrices=full_matrices)
+
+
+_svd_doc = """
+    Returns the SVD of a matrix or batch of matrices.
+
+    The `full_matrices` flag controls whether the full or reduced SVD decomposition is returned.
```
Comment (mruberry) [inline torch/_prims/__init__.py:2480]: Future question: should the full and reduced SVD be different prims?

Comment (Lezcano) [reply torch/_prims/__init__.py:2480]: Probably not. If we're going in the direction of ending up exposing something that smells like the LAPCK / cuSolver API and then write the dispatch in Python, we probably want them to be the same operation

**Review by mruberry — APPROVED** [review]
Cool! How, exactly, we model these prims is an interesting discussion for the future

**Pull request closed without merging**